src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -174,13 +181,14 @@ class CollectorService(
             val endTime = task.endTime
             when {
                 !olderThan(now, delta, unit, endTime) -> false
-                task.tryShutdownExecutor() -> {
-                    logger.info("Removed task ${task.description} ($endTime) from tasks map")
-                    true
-                }
                 else -> {
-                    logger.warn("Task ${task.description} can't be removed because it has a continuation")
-                    false
+                    !task.hasNextRule().also { canBeRemoved ->
+                        if (canBeRemoved) {
+                            logger.info("Removed task ${task.description} ($endTime) from tasks map")
+                        } else {
+                            logger.warn("Task ${task.description} can't be removed because it has a continuation")
```

Comment:
```suggestion
                        when {
                            canBeRemoved -> logger.info("Removed task ${task.description} ($endTime) from tasks map")
                            else -> logger.warn("Task ${task.description} can't be removed because it has a continuation")
```